[zh-tw]: translate HTTP `Session` (#19823)
mdn/translated-content: files/zh-tw/web/http/session/index.md

---
title: 典型的 HTTP 會話
slug: Web/HTTP/Session
l10n:
  sourceCommit: 46a16cc28c5da14b45a6f65e478d032532764122
---

> 

Reviewer: ```suggestion

{{HTTPSidebar}}

```

> 在用戶端-伺服器協定（例如 HTTP）中，會話由三個階段組成：

Reviewer: 下面的也用全角的吧，半角的影响阅读

![image](https://github.com/mdn/translated-content/assets/15844309/8f2ebac6-8ccd-41a2-94f2-9ba142636fd5)

```suggestion
在用戶端—伺服器協定（例如 HTTP）中，會話由三個階段組成：
```

---
title: 典型的 HTTP 會話
slug: Web/HTTP/Session
l10n:
  sourceCommit: 46a16cc28c5da14b45a6f65e478d032532764122
---

{{HTTPSidebar}}

在用戶端—伺服器協定（例如 HTTP）中，會話由三個階段組成：

1. 用戶端建立一個 TCP 連接（或者如果傳輸層不是 TCP，則建立適當的連接）。
2. 用戶端發送其請求，並等待回答。
3. 伺服器處理請求，發送其回答，提供狀態碼和適當的數據。

從 HTTP/1.1 開始，在完成第三個階段後不再關閉連接，並且現在用戶端被授予進一步的請求：這意味著第二和第三個階段現在可以執行任意次數。

## 建立連接

在用戶端—伺服器協定中，是用戶端建立連接。在 HTTP 中打開連接意味著在底層傳輸層啟動一個連接，這通常是 TCP。

> 對於 TCP，用於計算機上的 HTTP 伺服器的默認端口是 80。也可以使用其他端口，例如 8000 或 8080。要提取的頁面的 URL 包含域名和連接埠，如果連接埠是 80 則後者可以省略。有關更多詳細訊息，請參見[在 Web 上識別資源](/zh-TW/docs/Web/HTTP/Basics_of_HTTP/Identifying_resources_on_the_Web)。

Reviewer: ```suggestion
對於 TCP，用於計算機上的 HTTP 伺服器的默認端口是 80。也可以使用其他端口，例如 8000 或 8080。要提取的頁面的 URL 包含域名和連接埠，如果連接埠是 80 則後者可以省略。有關更多詳細訊息，請參見[識別 Web 上的資源](/zh-TW/docs/Web/HTTP/Basics_of_HTTP/Identifying_resources_on_the_Web)。
```

> **備註：** 用戶端—伺服器模型不允許伺服器在沒有明確請求的情況下向用戶端發送數據。但是，各種 Web API 使這種用例成為可能，包括 [Push API](/zh-TW/docs/Web/API/Push_API)、[伺服器發送事件](/zh-TW/docs/Web/API/Server-sent_events)和 [WebSockets API](/zh-TW/docs/Web/API/WebSockets_API)。

## 發送用戶端請求

> 一旦建立了連接，用戶代理就可以發送請求（用戶代理通常是網頁瀏覽器，但也可以是其他任何東西，例如爬蟲）。用戶端請求由文本指令組成，用 CRLF（回車符，後跟換行符）分隔，分成三個塊：

Reviewer: ```suggestion
一旦建立了連接，使用者代理就可以發送請求（使用者代理通常是網頁瀏覽器，但也可以是其他任何東西，例如爬蟲）。用戶端請求由文本指令組成，用 CRLF（回車符，後跟換行符）分隔，分成三個塊：
```

1. 第一行包含一個請求方法，後面跟著它的參數：

   - 文件的路徑，作為不帶協定或域名的絕對 URL
   - HTTP 協定版本

> 2. 接下來的行代表一個 HTTP 標頭，給伺服器提供關於哪種類型的數據是適合的訊息（例如，什麼語言，什麼 MIME 類型），或者其他改變其行為的數據（例如，如果已經緩存，則不發送答案）。這些 HTTP 標頭形成一個塊，以空行結束。

Reviewer: ```suggestion
2. 接下來的行代表一個 HTTP 標頭，給伺服器提供關於哪種類型的數據是適合的訊息（例如，什麼語言、什麼 MIME 類型），或者其他改變其行為的數據（例如，如果已經緩存，則不發送回應）。這些 HTTP 標頭形成一個塊，以空行結束。
```

3. 最後一個塊是一個可選的數據塊，其中可能包含主要由 POST 方法使用的進一步數據。

### 範例請求

> 提取 developer.mozilla.org 的根頁面（`https://developer.mozilla.org/`），並告訴伺服器用戶代理如果可能的話希望以法語顯示頁面：

Reviewer: ```suggestion
提取 developer.mozilla.org 的根頁面（`https://developer.mozilla.org/`），並告訴伺服器使用者代理如果可能的話希望以法語顯示頁面：
```
